### ISSUE TITLE
Dectection of bin packages is a bit weird

### ISSUE THREAD

**Issue by andoriyu** [nabijaczleweli/cargo-update#131]
From my understanding of code this packages traverse `~/.cargo/registry` to detect packages to update which is cool, but gives false negatives when binary has been deleted from the system, but it still cached in local registry.

Additionally, it leads to this scenario:
 - Install binary with `cargo install`
 - _time passes_
 - Delete the binary from `~/.bin`
 - Install binary as package from your OS package registry because that's the way
 - Now cargo-update will try to update things it should touch

I think it should check if binary found in path or even more specifically in cargo bin directory. Right now I have packages that fail to build 100% of time because they need to be built in a specific way and managed by OS package manager because of that.

**Comment by nabijaczleweli** [comment]
`cargo` stores the information about installed crates in `~/.cargo/.crates.toml`.
`cargo` installs binaries to `~/.cargo/bin`.
Neither `cargo` nor `cargo-update` touch `~/.bin`.
`cargo-update` does not traverse `~/.cargo/registry` in the manner implied.
`cargo-update` does not detect anything.
Moving executables installed by `cargo install` changes nothing in the operation of `cargo` and `cargo-update`.

**Comment by andoriyu** [comment]
> `cargo` installs binaries to `~/.cargo/bin`.
> Neither `cargo` nor `cargo-update` touch `~/.bin`.

Sorry, that's what I meant. 

> `cargo` stores the information about installed crates in `~/.cargo/.crates.toml`.

That makes much more sense that I what I thought is happening.

>  `cargo-update` does not detect anything.

This is actually the reason I opened this issue. Shouldn't it check if crate still installed?

**Comment by nabijaczleweli** [comment]
It reads the list of installed crates from `~/.cargo/.crates.toml`, any other sort of "checking" is not only physically impossible, but also kinda weird. I reiterate my last point:
Moving executables installed by `cargo install` changes nothing in the operation of `cargo` and `cargo-update`.

**Comment by andoriyu** [comment]
> It reads the list of installed crates from `~/.cargo/.crates.toml`, any other sort of "checking" is not only physically impossible, but also kinda weird. I reiterate my last point: Moving executables installed by `cargo install` changes nothing in the operation of `cargo` and `cargo-update` – are you actually uninstalling them or just (re)moving the executables?

Well, no. Up until your message I wasn't aware that cargo keeps track of things it installed.

> any other sort of "checking" is not only physically impossible, but also kinda weird

 Well, it can check if bins still in path? I see your point.

**Comment by nabijaczleweli** [comment]
What do we think of
```
>target\debug\cargo-install-update install-update -lag
    Polling registry 'https://index.crates.io/'.................

Package                          Installed    Latest   Needs update
bindgen-cli                      v0.71.1      v0.72.0  Yes
cargo-audit                      v0.20.0      v0.21.2  Yes
cargo-bisect-rustc               v0.6.9       v0.6.10  Yes
cargo-count                      v0.2.2       v0.2.4   Yes
cargo-deb                        v1.21.0      v2.12.1  Yes
cargo-graph                      v0.3.0       v0.3.1   Yes
cargo-outdated                   v0.3.0       v0.17.0  Yes
cargo-update                     v0.8.1       v16.3.1  Yes
checksums                        v0.5.3       v0.9.1   Yes
gen-epub-book                    v1.0.0       v2.3.2   Yes
identicon                        v0.1.1       v0.2.1   Yes
racer                            v2.1.42      v2.2.2   Yes
ripgrep                          v11.0.2      v14.1.1  Yes
tauri-cli                        v2.0.0-rc.3  v2.5.0   Yes
treesize                         v0.3.0       ^0.4     Yes
cargo-navigate                   v0.1.1       v0.1.1   No
rustup-toolchain-install-master  v1.8.0       v1.8.0   No

tauri-cli contains removed executables (tauri-cli.exe), which will be re-installed on update  run cargo uninstall tauri-cli to get rid of tauri-cli

    Checking 2 git packages..

Package  Installed                                 Latest                                    Needs update
https    152f4d70aba1db08fd047943f598e6d7bedb9a92  fb37da264e9c3463a3a06aef469b4af9cdb3443d  Yes
https2   b7deb3b46a7d16b2eaf1f8cfee040b5e1516b5d6  37f87c73181934233900c79985b91e54ee6a503c  Yes

https2 contains removed executables (httpgaming.exe), which will be re-installed on update  run cargo uninstall https2 to get rid of https2

```
?

**Comment by nabijaczleweli** [comment]
Released in [`v16.3.2`](https://github.com/nabijaczleweli/cargo-update/releases/tag/v16.3.2)
